### ISSUE TITLE
Add Pongo2 Template engine or another Django-syntax like template-engine for Go

### ISSUE THREAD

**Issue by bzerangue** [gohugoio/hugo#1359]
I would like to request that the Pongo2 Template Language be added as one of the templating languages that are available with Hugo. 

Pongo2 is a Django-syntax like template-engine for Go.

https://github.com/flosch/pongo2

This would be helpful for those switching from a Twig/Django/Liquid environment. Syntax very familiar. Also, it includes the very powerful `extends` template inheritance functionality.

**Comment by bep** [comment]
It looks nice and powerful and a well maintained project. But "adding it" to Hugo means work.

We no have 3: Go, Ace and Amber templates. The nice thing about Ace and Amber is that they extend on the template package in Go's stdlib, so when parsed "a template is just a template" and the different types can be mixed.

Pongo2 looks like "its own thing", which may make it harder. @tatsushid have worked more in this department, and might have some insight

**Comment by tatsushid** [comment]
I have also considered how about adding more template engines to Hugo and to make it easier, I have tried to refactor the Hugo's template library but it has been held up by the reason @bep pointed out that these engines syntax are not extensions of Go's standard but their own one.

It means, the variables and template functions which are now provided by Hugo may not work with these templates and it would need much effort to keep the template engine functionality working with Hugo, for example, including other template files into a caller template file. Not only the syntax, such a functionality is a heart of the engine so people who use the engine expect it is also supported in Hugo.

In addition, it would be difficult to help people who use the template engine which has its own world, a different syntax and a different way to access variables etc.

Above is just my opinion and I don't mean to object adding a new template engine but I would like to share my concern about it. Supporting a new template engine is great and I will reconsider there is a better way to do it.

**Comment by bzerangue** [comment]
Thank you both for the feedback. I didn't know if this was possible, I just thought I would ask. It would be great to see a Django-like syntax engine be apart of the Hugo ecosystem if it made sense for Hugo. Especially since both Amber and Ace are so similar as they are basically a Jade/Slim type template engine. It would be nice to have another familiar option. 

It's not a game-changer for me, but I do like the Django/Twig like markup and like the template inheritance functionality it utilizes. 

Just thought I would ask. I thank you for being gracious in your response and explaining why Pongo2 might not be an option. Very helpful to know the reasoning. Thank you again!

**Comment by ajslater** [comment]
Check this out: https://github.com/mattn/jedie
a pongo2 static site generator with blackfriday markdown support.

Much younger and more minimalist than hugo. Nowhere near as much sophisticated tooling. Little to no documentation to speak of. But that seems fine to me. Some of hugo's tooling looks like workarounds for go template limitations and I'm not interested in pre-built themes.

**Comment by codefx9** [comment]
@tatsushid / @bep. Do you mind pointing out how to go about adding a new template engine? I am familiar with Hugo as an user but I am not familiar with the codebase.

**Comment by bep** [comment]
@codefx9 if you this you mean adding Pongo2 or similar, I think you would have to get to know most of the parts of Hugo a little bit, and not just the template parts, as you would need to understand the big picture. Just start with the first package.

**Comment by moorereason** [comment]
You can probably start in `tpl/template.go` and look at the integration of the Ace and Amber template engines.

**Comment by bep** [comment]
This issue has been automatically marked as stale because it has not been commented on for at least six months.

The resources of the Hugo team are limited, and so we are asking for your help.

If this is a **bug** and you can still reproduce this error on the <code>master</code> branch, please reply with all of the information you have about it in order to keep the issue open.

If this is a **feature request**, and you feel that it is still relevant and valuable, please tell us why.

This issue will automatically be closed in the near future if no further activity occurs. Thank you for all your contributions.

**Comment by binyamin** [comment]
@bep Since Hugo no longer supports Ace or Amber, can this issue be reopened?

As a static-site builder, I find it odd that the only templating language Hugo supports is the one provided by Go. This essentially locks you into the Go ecosystem, which is less than ideal. Go templates can be confusing, since they're not similar to popular alternatives (eg. pug, jinja/nunjucks, django/liquid, mustache, handlebars).
